### PR TITLE
Add Auto Expiry Feature

### PR DESCRIPTION
**Pull request by mpchadwick**
Implementation of feature discussed in #110.

### CODE REVIEW

File: Cm/Cache/Backend/Redis.php [hunk 1]
```diff
@@ -700,6 +735,56 @@ public function setDirectives($directives)
...
+     * Mainly a workaround for the issues that arise due to the fact that
+     * Magento's Enterprise_PageCache module doesn't set any expiry.
+     *
+     * @param  int $specificLifetime
+     * @param  string $id
+     * @return int Cache life time
+     */
+    protected function _getAutoExpiringLifetime($lifetime, $id)
+    {
+        if ($lifetime) {
```
Comment (colinmollenhour) [inline Cm/Cache/Backend/Redis.php]: No reason to check if it matches the pattern if the feature is disabled. I recommend short-circuiting by changing this to:

```php
 if ($lifetime || ! $this->_autoExpireLifetime) {
```

File: Cm/Cache/Backend/Redis.php [hunk 2]
```diff
@@ -700,6 +735,56 @@ public function setDirectives($directives)
...
+
+    protected function _matchesAutoExpiringPattern($id)
+    {
+        $matches = array();
+        preg_match($this->_autoExpirePattern, $id, $matches);
+
+        return !empty($matches);
+    }
+
+    public function setAutoExpireLifetime($lifetime)
```
Comment (colinmollenhour) [inline Cm/Cache/Backend/Redis.php]: Other options are not exposed this way... Is there a reason they need to be set aside from the constructor?

File: Cm/Cache/Backend/Redis.php (hunk 1 above)
Comment (mpchadwick) [inline Cm/Cache/Backend/Redis.php]: Good call. Done.

File: Cm/Cache/Backend/Redis.php (hunk 2 above)
Comment (mpchadwick) [inline Cm/Cache/Backend/Redis.php]: The reason I added the setters was to be able to change the properties for `testAutoExpiry `. Maybe there's a better way, but I'm not sure. I tried Googling to see if `setUp` could tell what test method as being called, but couldn't find any information about that.

Comment (colinmollenhour) [inline Cm/Cache/Backend/Redis.php]: I think the proper way would be to have a new test case class for these tests that require a different setUp method.

Comment (mpchadwick) [inline Cm/Cache/Backend/Redis.php]: @colinmollenhour Updated with a separate test case. I'm following the pattern introduced with `forceStandalone` where a protected property is used and can be changed in the new test case class. Happy to squash these down as a final step if you'd like.

**Comment by colinmollenhour** [comment]
Looks good! Thanks, Max!

**Pull request merged**